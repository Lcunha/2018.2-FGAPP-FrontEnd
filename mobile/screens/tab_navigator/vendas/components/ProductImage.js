/*
  Component that renders a product (food, speacially) with name, price and image.
*/

import React, { Component } from 'react';
import { Image, View } from 'react-native';
import { Card, CardItem, Text, Left, Right } from 'native-base';
import styles from './styles';
import { LinearGradient } from 'expo';

class ProductImage extends Component {
  render() {
    return (
      <View style={{ flex: 1 }}>
        <Image
          style={ styles.image }
          source={{ uri: this.props.photo }}
        >
        </Image>
      </View>
    );
  }
}

export default ProductImage;
