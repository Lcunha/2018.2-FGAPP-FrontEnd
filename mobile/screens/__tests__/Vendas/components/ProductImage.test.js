import React from 'react';
import ProductImage from '../../../tab_navigator/vendas/components/ProductImage';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<ProductImage />).toJSON();
  expect(tree).toMatchSnapshot();
});
