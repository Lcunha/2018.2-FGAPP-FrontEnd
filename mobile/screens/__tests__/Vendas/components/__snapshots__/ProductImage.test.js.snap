// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 210,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 210,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <ExponentLinearGradient
      colors={
        Array [
          0,
          4278190080,
        ]
      }
      endPoint={undefined}
      locations={
        Array [
          0.989,
          0.9,
        ]
      }
      startPoint={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
